File: src/main/java/org/jhapy/notification/domain/CloudDataMessage.java
package org.jhapy.notification.domain;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;
import org.springframework.data.mongodb.core.mapping.Document;

/**
 * @author jHapy Lead Dev.
 * @version 1.0
 * @since 2019-07-09
 */
@Document
@Data
@EqualsAndHashCode(callSuper = true)
@ToString(callSuper = true)
public class CloudDataMessage extends BaseEntity {

  private String deviceToken;

  private String data;

  private String topic;

  private String cloudDataMessageAction;

  private CloudDataMessageStatusEnum cloudDataMessageStatus;

  private String errorMessage;

  private int nbRetry = 0;

  private String applicationName;

  private String iso3Language;
}

File: src/main/java/org/jhapy/notification/domain/CloudDataMessage.kt
package org.jhapy.notification.domain

import org.springframework.data.mongodb.core.mapping.Document

/**
 * @author jHapy Lead Dev.
 * @version 1.0
 * @since 2019-07-09
 */
@Document
class CloudDataMessage : BaseEntity() {
    var deviceToken: String? = null
    var data: String? = null
    var topic: String? = null
    var cloudDataMessageAction: String? = null
    var cloudDataMessageStatus: CloudDataMessageStatusEnum? = null
    var errorMessage: String? = null
    var nbRetry = 0
    var applicationName: String? = null
    var iso3Language: String? = null
}

enum class CloudDataMessageStatusEnum {
    NOT_SENT, SENT, ERROR, RETRYING
}